scripts/build_utils.sh

```diff
@@ -821,15 +821,28 @@ gen_debian_version() {
 # Flavor Builds support
 # - CEPH_EXTRA_RPMBUILD_ARGS is consumed by build_rpms()
 # - CEPH_EXTRA_CMAKE_ARGS is consumed by debian/rules and ceph.spec directly
-ceph_build_args_from_flavor() {
+ceph_build_args_from_flavor_and_dist() {
     local flavor=$1
     shift
+    local dist=$1
+    shift
 
     # shellcheck disable=SC2034
     case "${flavor}" in
     default)
         CEPH_EXTRA_RPMBUILD_ARGS="--with tcmalloc"
-        CEPH_EXTRA_CMAKE_ARGS+=" -DALLOCATOR=tcmalloc"
+        # https://github.com/ceph/ceph/pull/41340,
+        if [ "$dist" == "bionic" ]; then
+            local ver
+            ver=$(git describe --abbrev=8 --match "v*" | sed s/^v// | cut -f1 -d'.')
+            # use libc allocator when building on quincy and up on bionic
+            if [ ${ver} -lt 17 ]; then
```

Comment:
Why are building quincy on bionic at all?  Do we expect to support quincy on bionic in 2022 (and beyond)?

Reply:
@idryomov see #1794

Reply:
I see, thanks.  Might be nice to add the explanation that you closed #1794 with to the above "use libc allocator ..." comment.

Reply:
And drop the stray comma after the PR link.

Reply:
i don't understand how closing #1794 is related with this very change. but i amend the comment with more background on why we need to build with ubuntu/bionic.

Reply:
Well, this change adds a special case for something that you yourself thought wasn't necessary until the need for continuing building quincy on bionic was cleared up in #1794.  No one else is going to have that context and if I hadn't asked #1794 wouldn't even be linked in this PR, so I think having an explanation is useful.

Reply:
@idryomov could you help me understand what else explanation is necessary in addition to the comment? the special case is to use libc allocator on ubuntu when building quincy and up. how is it related to the fact of closing #1794 ?

Reply:
Yes, I assumed that we don't need to build quincy on bionic, as were you until you opened #1794.  I'm pretty sure that many others assume the same.  So for someone looking through this script and encountering a special case for quincy on bionic (and people usually don't look at these scripts unless there is a build or infrastructure issue which is often urgent), I thought it would be nice to explain *why* quincy is being built on bionic in the first place.

Does that answer your question?

Reply:
sort of. i understand your point, but i don't agree with you. i think people would be more interested in

> why we have special case for not using tcmalloc on bionic

instead of 

> why are we stilling building on bionic

#1794 is related. but i don't think this change is responsible to explain to the audience all the related facts. what it should do, is to explain why we need this change. anyway, i believe that a change should be specific and should be focus on the **change** itself instead of being an encyclopedia on all related topics.

Reply:
True, but "why we have special case for not using tcmalloc on bionic" is explained in the commit message (you included a link to the tracker ticket, etc) whereas "why are we stilling building on bionic" wasn't explained anywhere.

Reply:
> whereas "why are we stilling building on bionic" wasn't explained anywhere.

because it is not very related to the change, otherwise i'd add it in the first revision. but i added it anyway as it helps to provide more context, and so i can get my change merged. again, this change does not change the build matrix of distro x ceph-release, but it changes the matrix of distro x ceph-release x allocator, that's why i didn't want to include related information in the change.